tests/MaxMind/Test/ServiceClientTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace MaxMind\Test;
+
+use Composer\CaBundle\CaBundle;
+use MaxMind\WebService\Client;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @coversNothing
+ */
+abstract class ServiceClientTest extends TestCase
+{
+    protected function createRequest(
+        $class,
+        $urlTail,
+        $requestContent,
+        $statusCode,
+        $contentType,
+        $responseBody,
+        $options = [],
+        $callsToRequest = 1
+    ) {
+        $userId = 1;
+        $licenseKey = 'abcdefghij';
+
+        $stub = $this->getMockForAbstractClass(
+            'MaxMind\\WebService\\Http\\Request'
+        );
+
+        $stub->expects($this->exactly($callsToRequest))
+            ->method('post')
+            ->with($this->equalTo(json_encode($requestContent)))
+            ->willReturn([$statusCode, $contentType, $responseBody]);
+
+        $factory = $this->getMockBuilder(
+            'MaxMind\\WebService\\Http\\RequestFactory'
+        )->getMock();
+
+        $host = isset($options['host']) ? $options['host'] : 'minfraud.maxmind.com';
+
+        $url = 'https://' . $host . '/minfraud/v2.0/' . $urlTail;
+
+        $headers = [
+            'Content-Type: application/json',
+            'Authorization: Basic '
+            . base64_encode($userId . ':' . $licenseKey),
+            'Accept: application/json',
+        ];
+
+        if (isset($options['caBundle'])) {
+            $caBundle = $options['caBundle'];
+        } else {
+            $caBundle = CaBundle::getSystemCaRootBundlePath();
+        }
+
+        $curlVersion = curl_version();
+        $factory->expects($this->exactly($callsToRequest))
+            ->method('request')
+            ->with(
+                $this->equalTo($url),
+                $this->equalTo(
+                    [
+                        'headers' => $headers,
+                        'userAgent' => 'minFraud-API/' . $class::VERSION
```

Comment:
I want to point out `$class::VERSION` here -- there's no corresponding `use` statement at the top, so I'm not sure how it works. Tests do cover the user agent field though, so it's working, but I'm not sure how.

Reply:
You are passing the fully qualified name.

Reply:
Ah, OK, thanks!